unsupervised_da_exp.py

```diff
@@ -81,19 +100,57 @@ def eval_source_only(model, test_ds, datasets, target_domain, num_layers,
     so_model = model.headless
   else:
     so_model = model.classifier
-  for ds in datasets:
-    # copying the source weights to target variables
-    _ = mt.copy_weights(so_model, source=ds,
-                        target=target_domain,
-                        num_layers=num_layers,
-                        shared=True)
-    scores = model.evaluate(test_ds)
-    mt.write_scores(datasets, scores, results_file, f_mode="a+",
-                    scores_name="Source only %s" % ds)
-
-
-def get_da_model(classifier, ds_list, losses, loss_weights, lr_schedule,
-                 da_params: DomainAdaptationParameters):
+
+  # copying the source weights to target variables
+  _ = mt.copy_weights(so_model, source=copy_from,
+                      target=target_domain,
+                      num_layers=num_layers,
+                      shared=True)
+  scores = model.evaluate(test_ds, steps=num_steps)
+  mt.write_scores(datasets, scores, results_file, f_mode="a+",
+                  scores_name="Source only %s" % copy_from)
+
+def compile_da_model(da_model, losses, loss_weights,
+                     da_params=DomainAdaptationParameters(),
+                     lr_schedule=1e-4):
+  cl_optimizer = tfk.optimizers.RMSprop(learning_rate=lr_schedule)
+  da_optimizer = tfk.optimizers.Adam(learning_rate=lr_schedule)
+
+  if "discr" in da_params.da_mode:
```

Comment:
Just wondering, is there some way to simplify this (maybe using dictionaries and passing arguments through **kwargs)?

Reply:
good point; done in the following commit.